ua/expanded_node_id.go

```diff
@@ -24,8 +25,35 @@ type ExpandedNodeID struct {
 	ServerIndex  uint32
 }
 
+// Returns an OPC UA standard compliant string representation of the ExpandedNodeId,
+// described in Part 6, v1.0.5, 5.1.12 QualifiedName, NodeId and ExpandedNodeId String Encoding
+// The namespaceUri has to be URL encoded when the ExpandedNodeId is created
 func (a ExpandedNodeID) String() string {
-	return a.NodeID.String()
+	var s string = ""
+	if a.ServerIndex != uint32(0) {
+		s = fmt.Sprintf("svr=%d;", a.ServerIndex)
+	}
+	if len(a.NamespaceURI) > 0 {
+		s = fmt.Sprintf("%snsu=%s;", s, a.NamespaceURI)
+	} else {
+		if a.NodeID.ns != uint16(0) {
```

Comment:
same here with `uint16(0)`. 

Generally, I try to avoid `else`. How about this:

```go
switch {
    case len(a.NamespaceURI) > 0:
        s = fmt.Sprintf("%snsu=%s;", s, a.NamespaceURI)
    case a.NodeID.ns != 0:
        s = fmt.Sprintf("%sns=%d;", s, a.NodeID.ns)
}
```